### PR TITLE
Adding student councelor information for student

### LINKED ISSUE #5236: Add study guider information for student to see
HD Ticket #741964

### PR DESCRIPTION
**Pull request by jjlankinen**
Closes #5236

### CODE REVIEW

**Review by onzag — DISMISSED** [review]
Don't see any problems

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/records/summary.ts [hunk 1]
```diff
@@ -36,9 +36,30 @@ let updateSummary: UpdateSummaryTriggerType = function updateSummary() {
...
+      let studentsStudentCouncelors: any = [];
+
+      /*
+        We need to filter student's usergroups that are guidance groups, then we fetch guidance councelors
+        of those usergroups and push the result to studentsGuidanceCouncelors array
+      */
+      if (studentsUserGroups && studentsUserGroups.length) {
+        studentsUserGroups.filter((studentsUserGroup: any) => studentsUserGroup.guidanceGroup == true).forEach(function (studentsUserGroup: any) {
+          mApi().usergroup.groups.staffMembers.read(studentsUserGroup.id, {properties: 'profile-phone,profile-vacation-start,profile-vacation-end'}).callback(function (err: any, result: any) {
+            result.forEach(function (studentsStudentCouncelor: any) {
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/records/summary.ts:54]: The result set could probably be sorted by last name so the names wouldn't be in random order in the UI.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/summary.tsx [hunk 2]
```diff
@@ -47,6 +53,53 @@ class Summary extends React.Component<SummaryProps, SummaryState> {
...
               this.props.i18n.time.format(this.props.summary.data.studentsDetails.studyEndDate || this.props.summary.data.studentsDetails.studyTimeEnd) : this.props.i18n.text.get("plugin.records.summary.studyTime.empty")}</span></div>
           </div>
+
+          <div className="application-sub-panel__item">
+            <div className="application-sub-panel__item-title">{this.props.i18n.text.get("plugin.records.studyStudentCouncelorsLabel")}</div>
+            <div className="application-sub-panel__item-data application-sub-panel__item-data--summary-student-councelors">
+              <div className="item-list item-list--student-councelors">
+                {this.props.summary.data.studentsStudentCouncelors.map((councelor: SummaryStudentCouncelorsType, index: number) => {
+
+                  let displayVacationPeroid = !!councelor.properties['profile-vacation-start'];
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/summary.tsx:63]: displayVacationPeriod

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserGroupRESTService.java [hunk 3]
```diff
@@ -215,6 +228,64 @@ public Response listGroupUsersByGroup(@PathParam("ID") Long groupId) {
...
+    SchoolDataIdentifier loggedUser = sessionController.getLoggedUser();
+    UserGroupEntity userGroupEntity = userGroupEntityController.findUserGroupEntityById(groupId);
+    
+    if (userGroupEntity == null || Boolean.TRUE.equals(userGroupEntity.getArchived())) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    if (!sessionController.hasEnvironmentPermission(MuikkuPermissions.LIST_USERGROUP_STAFFMEMBERS)) {
+      if (!userGroupEntityController.isMember(loggedUser, userGroupEntity)) {
+        return Response.status(Status.NOT_FOUND).build();
```
Comment (pkukkon) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserGroupRESTService.java:244]: Shouldn't this be Status.FORBIDDEN instead?

**Review by pkukkon — CHANGES_REQUESTED** [review]

**Review by pkukkon — APPROVED** [review]

**Pull request merged**